### PR TITLE
CSI 1377 external csi pb2

### PR DESCRIPTION
**Pull request by ArbelNathan**

### CODE REVIEW

File: Dockerfile-csi-controller [hunk 1]
```diff
@@ -13,15 +13,21 @@
...
 USER default
 RUN pip3 install --ignore-installed --default-timeout=100 --upgrade pip==21.2.4
 # avoid default boringssl lib, since it does not support z systems
 ENV GRPC_PYTHON_BUILD_SYSTEM_OPENSSL=True
+COPY controller/requirements.txt /tmp/
 RUN pip3 install -r /tmp/requirements.txt
 
+USER root
+COPY controller/scripts/csi_general .
+RUN chmod 755 ./csi_pb2.sh
```
Comment (oriyarde) [inline Dockerfile-csi-controller:27]: is 755 the lowest value that works?
could it be submitted that way to begin with? I think it's currently with 644

File: Dockerfile-csi-controller [hunk 2]
```diff
@@ -13,15 +13,21 @@
...
+RUN ./csi_pb2.sh
+RUN pip3 install .
+USER default
```
Comment (oriyarde) [inline Dockerfile-csi-controller:30]: I don't think this is necessary

Comment (ArbelNathan) [reply Dockerfile-csi-controller:31]: so why requirements installed again?

Comment (oriyarde) [reply Dockerfile-csi-controller:31]: not sure what you mean. there is no need to install the requirements again.
I just don't see why we need the default user here

Comment (ArbelNathan) [reply Dockerfile-csi-controller:31]: the conversation is only on the  `USER default`?

Comment (oriyarde) [reply Dockerfile-csi-controller:31]: yes...

Comment (ArbelNathan) [reply Dockerfile-csi-controller]: done

File: Dockerfile-csi-controller (hunk 1 above)
Comment (ArbelNathan) [reply Dockerfile-csi-controller]: seems that it worked
done

File: Dockerfile-csi-controller.test [hunk 3]
```diff
@@ -17,18 +17,24 @@
 #    Its similar to the Dockerfile, but with additional requirements-tests.txt and ENTRYPOINT to run the local tests.
 
 FROM registry.access.redhat.com/ubi8/python-38:1-75.1638364053 as builder
-COPY controller/requirements.txt /tmp/
 USER root
 RUN if [[ "$(uname -m)" != "x86"* ]]; then yum install -y rust-toolset; fi
 USER default
 RUN pip3 install --ignore-installed --default-timeout=100 --upgrade pip==21.2.4
 # avoid default boringssl lib, since it does not support z systems
 ENV GRPC_PYTHON_BUILD_SYSTEM_OPENSSL=True
```
Comment (oriyarde) [inline Dockerfile-csi-controller.test:25]: I think these two lines should come right before the `pip3 install`

**Review by oriyarde — APPROVED** [review]
LGTM

File: Dockerfile-csi-controller.test (hunk 3 above)
Comment (ArbelNathan) [reply Dockerfile-csi-controller.test]: done

**Pull request merged**